docs/content/concepts/transactions/WIP-ptb.md

```diff
@@ -0,0 +1,400 @@
+
+This gives us updated memory locations of
+```
+Gas Coin: _
+Inputs: [
+  Pure(/* @0x808 BCS bytes */ ...),
+  Marketplace { id: market_id, ... },
+  Pure(/* 100u64 BCS bytes */ ...),
+]
+Results: [
+  [ _ ],
+  [ _ , Item { id: id2 }],
+  [],
+  [/* senders address */ ...], // The result of the Move call
+],
+```
+
+### Command 4: `TransferObjects`
+
+Finally, we will transfer the remaining item to ourselves. This is similar to the previous `TransferObjects` command. We are using the last `Item` by-value and the sender's address by-value. The item is moved since `Item` does not have `copy`, and the address is copied since `address` does.
+
+The final memory locations are
+```
+Gas Coin: _
+Inputs: [
+  Pure(/* @0x808 BCS bytes */ ...),
+  Marketplace { id: market_id, ... },
+  Pure(/* 100u64 BCS bytes */ ...),
+]
+Results: [
+  [ _ ],
+  [ _ , _ ],
+  [],
+  [/* senders address */ ...],
+  [], // No results from TransferObjects
+],
+```
+
+### After Commands: End of Execution
+
+At the end of execution, the runtime checks the remaining values, which are the 3 inputs and
+the sender's address. The following summarizes the checks performed before effects are given:
+
+- Any remaining input objects are marked as being returned to their original owners.
+  - The gas coin has been Moved. And the `Marketplace` will keep the same owner, which is shared.
+- All remaining values must have `drop`.
+  - The Pure inputs have `drop` since any type they can instantiate has `drop`.
+  - The sender's address has `drop` since it the primitive type `address` has `drop`.
+  - All other results have been moved.
+- Any remaining shared objects must have been deleted or re-shared.
+  - The `Marketplace` object was not moved, so the owner remains as shared.
+
+After these checks are performed, we generate the effects.
+
+- The coin split off from the gas coin, `new_coin`, does _not_ appear in the effects since it was created and deleted in the same transaction.
+- The gas coin and the item with `id1` are transferred to `0x808`.
+  - The gas coin is mutated to update its balance. The remaining gas of the maximum budget of `500_000` is returned the gas coin even though the owner has changed.
+  - The `Item` with `id1` is a newly created object.
+- The item with `id2` is transferred to the sender's address.
+  - The `Item` with `id2` is a newly created object.
+- The `Marketplace` object is returned remains shared and its mutated.
```

Comment:
```suggestion
- The `Marketplace` object is returned, remains shared, and it's mutated.
```
I am not sure if I got the intent of this sentence right, though...

Reply:
I think you did. Thanks!